src/cml.js

```diff
@@ -143,59 +143,57 @@ class CML {
     return await getDriver(this).runnerToken();
   }
 
-  parseRunnerLog(opts = {}) {
+  async parseRunnerLog(opts = {}) {
     let { data } = opts;
     if (!data) return;
 
+    const date = new Date();
+
     try {
       data = data.toString('utf8');
 
       let log = {
         level: 'info',
-        time: new Date().toISOString(),
+        date: date.toISOString(),
         repo: this.repo
       };
 
       if (this.driver === GITHUB) {
         if (data.includes('Running job')) {
-          log.job = '';
+          const { id } = await getDriver(this).job({ time: date.getTime() });
+          log.job = id;
           log.status = 'job_started';
-          return log;
         } else if (
           data.includes('Job') &&
           data.includes('completed with result')
         ) {
           log.job = '';
           log.status = 'job_ended';
-          log.success = data.endsWith('Succeeded');
+          log.success = data.includes('Succeeded');
           log.level = log.success ? 'info' : 'error';
-          return log;
         } else if (data.includes('Listening for Jobs')) {
           log.status = 'ready';
-          return log;
         }
+        return log;
       }
 
       if (this.driver === GITLAB) {
         const { msg, job } = JSON.parse(data);
+        log = { ...log, job };
 
         if (msg.endsWith('received')) {
-          log = { ...log, job };
           log.status = 'job_started';
-          return log;
         } else if (
           msg.startsWith('Job failed') ||
           msg.startsWith('Job succeeded')
         ) {
-          log = { ...log, job };
           log.status = 'job_ended';
           log.success = !msg.startsWith('Job failed');
           log.level = log.success ? 'info' : 'error';
-          return log;
         } else if (msg.includes('Starting runner for')) {
           log.status = 'ready';
-          return log;
         }
+        return log;
```

Comment:
Do these improvements belong to this pull request? If so, can you please add some unit tests with the original runner messages?

Maybe it's just nitpicking, but the `parseRunnerLog` function is crucial for status checking and, if it fails, it could produce some creative errors. 😄

Reply:
Actually fixes a bug and then it just only introduces less returns

Reply:
Did I got a 👍 here? And then you removed it? It usually happens to me with you. You are the master of ephemeral emojis

Reply:
Perhaps. 😄  I probably added it because you clarified the intent of your modifications, and then removed it hoping that you would post another message after adding a few unit tests and explained what's the bug you were fixing.

My emoji policy has more exceptions than your linter configuration. 😉  I'll have to document all them somewhere.